packages/xo-web/src/common/sorted-table/index.js

```diff
@@ -209,29 +209,23 @@ class IndividualAction extends Component {
       isFunction(disabled) ? disabled(item, userData) : disabled
   )
 
-  _redirect = createSelector(
-    () => this.props.item,
-    () => this.props.redirect,
-    () => this.props.userData,
-    (item, redirect, userData) =>
-      isFunction(redirect) ? redirect(item, userData) : redirect
-  )
-
   _executeAction = () => {
     const p = this.props
-    return p.handler && p.handler(p.item, p.userData)
+    return p.redirectOnSuccess
+      ? { item: p.item, userData: p.userData }
```

Comment:
@pdonias  you think it's better to use `handlerParam` & add this change on `ActionButton` :  
```js
      const result = await handler(handlerParam)

      const { redirectOnSuccess } = props
      if (redirectOnSuccess) {
        return this.context.router.push(
          isFunction(redirectOnSuccess)
            ? result
              ? redirectOnSuccess(result)
              : redirectOnSuccess(handlerParam[`item`], handlerParam[`userData`])
            : redirectOnSuccess
        )
      }
      ```